src/realm/object_converter.cpp

```diff
@@ -368,8 +370,21 @@ int InterRealmValueConverter::cmp_src_to_dst(Mixed src, Mixed dst, ConversionRes
             }
         }
         else {
-            Mixed src_link_pk = m_opposite_of_src->get_primary_key(src_link_key);
-            Obj dst_link = m_opposite_of_dst->create_object_with_primary_key(src_link_pk, did_update_out);
+            Obj dst_link;
+            if (m_opposite_of_src->get_primary_key_column()) {
+                Mixed src_link_pk = m_opposite_of_src->get_primary_key(src_link_key);
+                dst_link = m_opposite_of_dst->create_object_with_primary_key(src_link_pk, did_update_out);
+            }
+            else {
+                if (m_opposite_of_dst == m_opposite_of_src) {
+                    // if this is the same Realm, we can use the ObjKey
+                    dst_link = m_opposite_of_dst->get_object(src_link_key);
+                }
+                else {
+                    // in different Realms we create a new object
```

Comment:
Is it relevant to support a case where we copy between 2 different realms in case the object does not have a primary key? I guess it will be hard to create a test case for this?

Reply:
I would not add a test for this right now, I don't think this could happen within the context of a schema migration.